File: src/ios/ScannerAction.swift
//
//  ScannerAction.swift
//  MLBarcodeScanner
//
//  Created by Felix Nievelstein on 24.11.21.
//

import Foundation

enum ScannerAction: String {
    case IS_VALID = "isValid"
    case RESULT_SCREEN = "resultScreen"
    case DID_SELECT_CODE = "didSelectCode"
    case MANUAL_INPUT = "preferredManuelInput"
    case WILL_CLOSE = "willClose"
    case DID_CLOSE = "didClose"
}

File: src/android/ScannerAction.kt
package de.impacgroup.mlbarcodescanner

enum class ScannerAction(val key: String) {
    IS_VALID("isValid"),
    RESULT_SCREEN("resultScreen"),
    DID_SELECT_CODE("didSelectCode"),
    MANUAL_INPUT("preferredManuelInput"),
    WILL_CLOSE("willClose"),
    DID_CLOSE("didClose"),
    UNKNOWN("unknown");

    companion object {
        fun getFor(name: String): ScannerAction {
            return when(name) {
                DID_CLOSE.key -> DID_CLOSE
                WILL_CLOSE.key -> WILL_CLOSE
                IS_VALID.key -> IS_VALID
                DID_SELECT_CODE.key -> DID_SELECT_CODE
                MANUAL_INPUT.key -> MANUAL_INPUT
                RESULT_SCREEN.key -> RESULT_SCREEN
                else -> UNKNOWN
            }
        }
    }
}